wcivf/apps/elections/models.py

```diff
@@ -468,6 +468,16 @@ def next_ballot(self):
         except PostElection.DoesNotExist:
             return None
 
+    @property
+    def party_ballot_count(self):
+        people = self.people
+        independent_candidates = people.filter(party_id="ynmp-party:2").count()
+        num_all_parties = people.values("party_id").distinct().count()
+        if self.election.uses_lists:
+            return len(independent_candidates) + num_all_parties
+        else:
+            return num_all_parties
```

Comment:
This is looking good with some small changes:

1. In the `else` we actually want to just return the number of people (`people.count()`) rather than the number of parties
1. Because of this, `num_all_parties` and `independent_candidates` can be moved inside the if condition

This means that if we have lists in use we count the distinct parties and add the independents, if lists not in use we just count the people.

One untested thing: I think `ynmp-party:2` will be included in `num_all_parties`, so you'll want to exclude that to save getting one more than we should.